src/sentry/auth/authenticators/u2f.py

```diff
@@ -116,8 +136,34 @@ def activate(self, request):
 
         return ActivationChallengeResult(challenge=challenge)
 
-    def validate_response(self, request, challenge, response):
+    def validate_response(self, request, challenge, response, is_webauthn_signin_ff_enabled):
         try:
+            if is_webauthn_signin_ff_enabled:
+                # TODO change rp.id later when register is implemented
+                server = U2FFido2Server(
+                    app_id=challenge["appId"],
+                    rp={"id": challenge["appId"], "name": "Relying Party"},
+                )
+                state = {
+                    "challenge": challenge["challenge"],
+                    "user_verification": None,
+                }
+                credentials = []
+                for registeredKey in challenge["registeredKeys"]:
+                    c = base.AttestedCredentialData.from_ctap1(
+                        websafe_decode(registeredKey["keyHandle"]),
+                        websafe_decode(registeredKey["publicKey"]),
+                    )
+                    credentials.append(c)
```

Comment:
i would move this into its own function

Reply:
I think i realized this and move this into its own function in the following pr for registration